### PR TITLE
Table Error Handling and Schema

### PR DESCRIPTION
**Pull request by AndreaLandiArk**
- Added Zod schema for header building in tables.
- Added better error handling for tables

### CODE REVIEW

File: src/components/tables/plainTable/chackraPlainTable.tsx [hunk 1]
```diff
@@ -1,73 +1,85 @@
...
+      // you can access a string 'message' property here
+      details = {
+        title: "Fetching Error",
+        message: error.message,
+        isValidationError: false,
+        exceptionDetails: null,
+      } as DetailsType;
+    }
+    setError({ error: true, details });
+    return <></>;
```
Comment (AndreaCuneo) [inline src/components/tables/plainTable/chackraPlainTable.tsx:49]: I'm not sure setting the global error is the right approach and not to display the Table, thus changing a lot the page display.

1. Ensure that on fetch error, the display below the error popup is ok
2. Should in this case the error have a 'reload' or else, after an error, how the page can recover without a full SPA reload!
3. Errors in background polling (refresh) not sure should break the flow. The `isLoading` fair currently doesn't toggle on background polling (`isFetching` is for it) and the spinner do not replace the table during polling. Same for errors. Afair you can have both {data} and {error} in that case. Please check rtkq docs.

**Review by AndreaCuneo — CHANGES_REQUESTED** [review]

File: src/components/tables/plainTable/chackraPlainTable.tsx (hunk 1 above)
Comment (AndreaLandiArk) [reply src/components/tables/plainTable/chackraPlainTable.tsx]: Now the table is still visible and if an error occour after the first fetch (while polling), in the example, we show a toast.
Added a comment to it also.
Checking isFetching other than isLoading is page duty to me so I added it there.
Added a reload page button on the error modal

File: src/components/problemDetailsModal/problemDetailsModal.tsx [hunk 2]
```diff
@@ -46,15 +47,11 @@ const ProblemDetailsModalBody = (props: { problem: DetailsType | null }) => {
...
-            {problem?.originalDetail
-              ? problem.originalDetail
-              : problem?.message}
-          </Code>
+          <Code colorScheme="grey">{problem?.originalDetail ? problem.originalDetail : problem?.message}</Code>
         </Card>
       </Flex>
       <Flex my="20px">
-        <Accordion>
+        <Accordion w="-webkit-fill-available">
```
Comment (AndreaCuneo) [inline src/components/problemDetailsModal/problemDetailsModal.tsx:54]: ?

Is this a chakra feature?
I think something like "full" should be used instead.
Or maybe is mapped by chakra on the size prop.

**Review by AndreaCuneo — CHANGES_REQUESTED** [review]

File: src/features/fetchApiExample/jsonPlaceholderSlice.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ import type { PostDataType } from "./jsonPlaceholderTypes";
 export const jsonPlaceholderSlice = createApi({
   reducerPath: "jsonPlaceholder",
   baseQuery: fetchBaseQuery({
-    baseUrl: "https://jsonplaceholder.typicode.com/",
+    baseUrl: "https://jsonplaceholdes.typicode.com/",
```
Comment (AndreaCuneo) [inline src/features/fetchApiExample/jsonPlaceholderSlice.ts:7]: Still wrong. Have you tested this change?

**Review by AndreaCuneo — CHANGES_REQUESTED** [review]

**Pull request merged**